addons/nuxeo-retention/nuxeo-retention-core/src/test/java/org/nuxeo/retention/test/TestRetentionEvents.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * (C) Copyright 2020 Nuxeo (http://nuxeo.com/) and others.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Salem Aouana
+ */
+
+package org.nuxeo.retention.test;
+
+import static org.junit.Assert.assertArrayEquals;
+import static org.junit.Assert.assertEquals;
+import static org.nuxeo.ecm.core.api.CoreSession.RETAIN_UNTIL_INDETERMINATE;
+
+import java.util.Calendar;
+
+import javax.inject.Inject;
+
+import org.junit.Test;
+import org.nuxeo.ecm.core.api.event.DocumentEventTypes;
+import org.nuxeo.ecm.core.event.Event;
+import org.nuxeo.ecm.core.event.test.CapturingEventListener;
+import org.nuxeo.runtime.test.runner.TransactionalFeature;
+
+/**
+ * @since 11.1
+ */
+public class TestRetentionEvents extends RetentionTestCase {
+
+    @Inject
+    protected TransactionalFeature transactionalFeature;
+
+    @Test
+    public void shouldNotifyEventsWhenModifyLegalHold() {
+        session.makeRecord(file.getRef());
+
+        checkLegalHoldEvents(true, "I put the legal hold", DocumentEventTypes.BEFORE_SET_LEGAL_HOLD,
+                DocumentEventTypes.AFTER_SET_LEGAL_HOLD);
+
+        checkLegalHoldEvents(false, "I remove the legal hold", DocumentEventTypes.BEFORE_REMOVE_LEGAL_HOLD,
+                DocumentEventTypes.AFTER_REMOVE_LEGAL_HOLD);
+
+    }
+
+    protected void checkLegalHoldEvents(boolean hold, String comment, String... expectedEvents) {
```

Comment:
I would go for `setLegalHoldAndCheckEvents` because this method doesn't just check events.